src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -981,8 +981,7 @@ public function __construct()
         if ($this->container->isCompiled()) {
             if (Container::class !== $baseClassWithNamespace) {
                 $r = $this->container->getReflectionClass($baseClassWithNamespace, false);
-
-                if (null !== $r && (null !== $constructor = $r->getConstructor()) && 0 === $constructor->getNumberOfRequiredParameters()) {
+                if (null !== $r && (null !== $constructor = $r->getConstructor()) && 0 === $constructor->getNumberOfRequiredParameters() && $constructor->getDeclaringClass() === Container::class) {
```

Comment:
should be `!==`, not `===` We want to call the parent constructor when it is a custom one.

Reply:
that doesn't fix the issue .. since in the MockContainer the declaring class of the constructor is the Container class, since the MockContainer doesn't define its own constructor

Reply:
and that's precisely the goal: we **don't** want to call the parent constructor for the MockContainer case (otherwise, you would not have a bug currently when it is called).

The Container constructor initializes the ParameterBag, and the dumped container wants to do it in its own way.

Reply:
but in that case we don't want to call the parent .. 
if I do `!==` the bug remains .. I have confirmed this, because we still end up calling the parent constructor.

Reply:
your code is adding the `parent::__construct` *only* when the declaring class of the constructor is Container now.

Reply:
ah you are right .. the issue was that it needed to be `$constructor->getDeclaringClass()->name` and not `$constructor->getDeclaringClass()` in the comparison.